### ISSUE TITLE
Edit software

### ISSUE THREAD

**Issue by goldandib** [Entkenntnis/create-challenges-server#5]
I have installed the software on Glitch and I wonder how I can change the code of the Challenges Server. It seems that the code is requested everytime from GitHub because I don't see much code in the files in the Glitch project. Is there an easy way to use a custom version on Glitch?

**Comment by Entkenntnis** [comment]
Hi AndiDandi, glad that you are passing by!

The source code of the server is locating in this repository: https://github.com/Entkenntnis/challenges-server. You can change the dependency and point it to your fork.

That being said, the challenges-server now has some specifiy changes that are only relevant for https://hack.arrrg.de - if you encounter any such limitation, I'm happy to assist.

**Comment by goldandib** [comment]
I tried it out, but unfortunately it doesn't work as expected. It shows the error "cannot find module". I even tried to change my GitHub Username because I thought maybe the number in it could be the problem. Do you have an idea what the problem could be?
![Screenshot 2024-01-12 162039](https://github.com/Entkenntnis/create-challenges-server/assets/156343618/e09a67b5-903d-4951-bdcf-4b5e068eaadb)

**Comment by Entkenntnis** [comment]
Can you show me the package.json? You need to change the version to your repo url and then run npm install.

**Comment by goldandib** [comment]
Ah, I thought I only have to make a change in the index.js. Could you tell me what I have to change in the json? I have only forked the challenges-server repo so far, do I also have to create a fork of the create-challenges-server repo? Because the json also points at the create-challenges-server repo.
![Screenshot 2024-01-12 184000](https://github.com/Entkenntnis/create-challenges-server/assets/156343618/e052f840-706f-492c-acfb-ef0024b9dbc2)

**Comment by Entkenntnis** [comment]
changing line 16 to `"@entkenntnis/challenges-server": "goldandib/challenges-server"`, should do the trick

**Comment by goldandib** [comment]
I'm sorry it still doesn't work, although I have the original repo in my fork so far. It permanently shows this "Resolving: total 122, reused 122, downloaded 0" messages in the log and the npm install command takes over half an hour. I'm afraid I'm too unexperienced to do this, unfortunately.
![Screenshot 2024-01-14 204232](https://github.com/Entkenntnis/create-challenges-server/assets/156343618/2cde04d3-f835-4385-8c36-a7a8ca871b5f)

**Comment by Entkenntnis** [comment]
Hi goldandib, I tried it myself and encountered the same problem on glitch - I didn't manage to get it to work, too.

I created a new repo that includes the whole source code: https://github.com/Entkenntnis/challenges-server-starter.

Importing this into glitch makes it working. You can edit everything within the folder challenges-server:
![grafik](https://github.com/Entkenntnis/create-challenges-server/assets/13507950/110aef4e-08ee-4368-b54b-cb1372f02d3a)

**Comment by goldandib** [comment]
Thank you very much! That's a perfect solution.

**Comment by Entkenntnis** [comment]
I'm a little bit curious - so if you don't mind I'd be interested to hear how you encountered the challenges-server and what topic your project will have :)

**Comment by goldandib** [comment]
Yeah, I heared from HackTheWeb from my teacher and I started playing it. I have completed many challenges already (not all) (my name there is Goldandi) and I enjoyed playing it, because I want to get better in hacking, programming and these things. I could learn many things from it already. I also liked the game system itself, with the map and the challenges as points on it, and I think it would be perfect for a scavenger hunt, if you put a map in the background and then the stations on it as challenges where they are. I'm thinking about creating a scavenger hunt for my home town with it. And I also like to get more experienced with developing itself, it's a nice software, but it's not too complicated so I have a chance to understand how it works and then I can try to change things and see what effect it has. Some minor changes I already did successfully. I think it's more fun to learn developing if you first have something for playing around. I'm not sure yet if and when I do a public project online. When I have one online, I will let you know :)
